Crashpad compilation error with CMake 4 on macOS

`getsentry/crashpad` relies on [`CMAKE_OSX_SYSROOT`](https://cmake.org/cmake/help/latest/variable/CMAKE_OSX_SYSROOT.html) which is no longer set in CMake 4.0:

> _Changed in version 4.0_: The default is now empty. Previously a default was computed based on the [`CMAKE_OSX_DEPLOYMENT_TARGET`](https://cmake.org/cmake/help/latest/variable/CMAKE_OSX_DEPLOYMENT_TARGET.html#variable:CMAKE_OSX_DEPLOYMENT_TARGET) or the host platform.

With CMake 4.0, the error is:

```sh
$ cmake -B build
-- The C compiler identification is AppleClang 17.0.0.17000013
-- The CXX compiler identification is AppleClang 17.0.0.17000013
-- The ASM compiler identification is AppleClang
-- Found assembler: /usr/bin/cc
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- SENTRY_TRANSPORT=curl
-- SENTRY_BACKEND=crashpad
-- SENTRY_LIBRARY_TYPE=SHARED
-- SENTRY_SDK_NAME=
-- SENTRY_HANDLER_STACK_SIZE=64
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of long
-- Check size of long - done
-- Found CURL: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib/libcurl.tbd (found version "8.7.1")
-- Found ZLIB: /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib/libz.tbd (found version "1.2.12")
CMake Error at external/crashpad/util/CMakeLists.txt:416 (message):
  File not found: /usr/include/mach/exc.defs


-- Configuring incomplete, errors occurred!
```

Brew has had CMake 4.0 for a while now, so I've been using this as a temporary workaround:

```diff
diff --git a/util/CMakeLists.txt b/util/CMakeLists.txt
index b804b7ad..e5565a94 100644
--- a/util/CMakeLists.txt
+++ b/util/CMakeLists.txt
@@ -398,6 +398,9 @@ endif()

 # Copied from: https://github.com/qedsoftware/crashpad/blob/3583c50a6575857abcf140f6ea3b8d11390205b3/util/CMakeLists.txt#L196-L233
 if(APPLE)
+    if(NOT CMAKE_OSX_SYSROOT)
+        set(CMAKE_OSX_SYSROOT "/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk")
+    endif()
     if(NOT IOS)
         set(def_relative_files "exc.defs" "mach_exc.defs" "notify.defs")
         set(input_files "${CMAKE_CURRENT_LIST_DIR}/mach/child_port.defs")
```

--- a/util/CMakeLists.txt
+++ b/util/CMakeLists.txt
@@ -397,6 +397,19 @@ endif()
 # Copied from: https://github.com/qedsoftware/crashpad/blob/3583c50a6575857abcf140f6ea3b8d11390205b3/util/CMakeLists.txt#L196-L233
 if(APPLE)
     if(NOT IOS)
+        if(NOT CMAKE_OSX_SYSROOT)
+            # In CMake 4.0, CMAKE_OSX_SYSROOT defaults to empty unless SDKROOT is set:
+            # https://cmake.org/cmake/help/latest/variable/CMAKE_OSX_SYSROOT.html
+            execute_process(
+                COMMAND xcrun --sdk macosx --show-sdk-path
+                RESULT_VARIABLE CMAKE_OSX_SYSROOT_RESULT
+                OUTPUT_VARIABLE CMAKE_OSX_SYSROOT
+                OUTPUT_STRIP_TRAILING_WHITESPACE
+            )
+            if(NOT CMAKE_OSX_SYSROOT_RESULT EQUAL 0)
+                message(FATAL_ERROR "Failed to resolve CMAKE_OSX_SYSROOT. Please specify SDKROOT in the environment.")
+            endif()
+        endif()
         set(def_relative_files "exc.defs" "mach_exc.defs" "notify.defs")
         set(input_files "${CMAKE_CURRENT_LIST_DIR}/mach/child_port.defs")
     else()
@@ -407,8 +420,6 @@ if(APPLE)
         )
     endif()
     foreach(x ${def_relative_files})
-        # CMAKE_OSX_SYSROOT may be empty (e.g. for Makefile generators),
-        # in this case files will be taken from root.
         set(full_path "${CMAKE_OSX_SYSROOT}/usr/include/mach/${x}")
         if(NOT EXISTS "${full_path}")
             message(FATAL_ERROR "File not found: ${full_path}")

--- a/util/CMakeLists.txt
+++ b/util/CMakeLists.txt
@@ -396,23 +396,31 @@ endif()
 
 # Copied from: https://github.com/qedsoftware/crashpad/blob/3583c50a6575857abcf140f6ea3b8d11390205b3/util/CMakeLists.txt#L196-L233
 if(APPLE)
-    if(NOT IOS)
+    function(check_and_fallback_sdk_path SDK_NAME)
         if(NOT CMAKE_OSX_SYSROOT)
-            # In CMake 4.0, CMAKE_OSX_SYSROOT defaults to empty unless SDKROOT is set:
-            # https://cmake.org/cmake/help/latest/variable/CMAKE_OSX_SYSROOT.html
+            message("Empty CMAKE_OSX_SYSROOT. Please specify SDKROOT=${SDK_NAME} in the environment.")
+            message("Falling back to searching SDK path for `${SDK_NAME}` via `xcrun`.")
             execute_process(
-                COMMAND xcrun --sdk macosx --show-sdk-path
-                RESULT_VARIABLE CMAKE_OSX_SYSROOT_RESULT
-                OUTPUT_VARIABLE CMAKE_OSX_SYSROOT
+                COMMAND xcrun --sdk ${SDK_NAME} --show-sdk-path
+                RESULT_VARIABLE XCRUN_RESULT
+                OUTPUT_VARIABLE SDK_PATH
                 OUTPUT_STRIP_TRAILING_WHITESPACE
             )
-            if(NOT CMAKE_OSX_SYSROOT_RESULT EQUAL 0)
-                message(FATAL_ERROR "Failed to resolve CMAKE_OSX_SYSROOT. Please specify SDKROOT in the environment.")
+
+            if(NOT XCRUN_RESULT EQUAL 0)
+                message(FATAL_ERROR "Can't find ${SDK_NAME} SDK. `xcrun` failed with ${XCRUN_RESULT}.")
+            else()
+                message("Setting CMAKE_OSX_SYSROOT=${SDK_PATH}.")
+                set(CMAKE_OSX_SYSROOT "${SDK_PATH}" PARENT_SCOPE)
             endif()
         endif()
+    endfunction()
+    if(NOT IOS)
+        check_and_fallback_sdk_path(macosx)
         set(def_relative_files "exc.defs" "mach_exc.defs" "notify.defs")
         set(input_files "${CMAKE_CURRENT_LIST_DIR}/mach/child_port.defs")
     else()
+        check_and_fallback_sdk_path(iphoneos)
         set(def_relative_files "")
         set(input_files
             "${CMAKE_CURRENT_LIST_DIR}/../third_party/xnu/osfmk/mach/exc.defs"

